python/hail/expr/functions.py

```diff
@@ -3479,3 +3479,115 @@ def get_sequence(reference_genome, contig, position, before=0, after=0) -> Strin
     return _func("getReferenceSequence({})".format(reference_genome.name), tstr,
                  contig, position, before, after)
+
+
+@typecheck(locus=expr_locus(), is_female=expr_bool, father=expr_call, mother=expr_call, child=expr_call)
+def mendel_error_code(locus, is_female, father, mother, child):
+    """Compute a Mendelian violation code for genotypes.
+
+    >>> father = hl.call(0, 0)
+    >>> mother = hl.call(1, 1)
+    >>> child1 = hl.call(0, 1)  # consistent
+    >>> child2 = hl.call(0, 0)  # Mendel error
+    >>> locus = hl.locus('2', 2000000)
+
+    >>> hl.mendel_error_code(locus, True, father, mother, child1).value
+    None
+
+    >>> hl.mendel_error_code(locus, True, father, mother, child2).value
+    7
+
+    Note
+    ----
+    Ignores call phasing, and assumes diploid and biallelic.
+
+    Notes
+    -----
+    In the table, the copy state of a locus with respect to a trio is defined
+    as follows, where PAR is the `pseudoautosomal region
+    <https://en.wikipedia.org/wiki/Pseudoautosomal_region>`__ (PAR) of X and Y
+    defined by the reference genome and the autosome is defined by
+    :meth:`.LocusExpression.in_autosome`.
+
+    - Auto -- in autosome or in PAR, or in non-PAR of X and female child
+    - HemiX -- in non-PAR of X and male child
+    - HemiY -- in non-PAR of Y and male child
+
+    `Any` refers to the set \{ HomRef, Het, HomVar, NoCall \} and `~`
+    denotes complement in this set.
+
+    +------+---------+---------+--------+----------------------------+
+    | Code | Dad     | Mom     | Kid    | Copy State | Implicated    |
+    +======+=========+=========+========+============+===============+
+    |    1 | HomVar  | HomVar  | Het    | Auto       | Dad, Mom, Kid |
+    +------+---------+---------+--------+------------+---------------+
+    |    2 | HomRef  | HomRef  | Het    | Auto       | Dad, Mom, Kid |
+    +------+---------+---------+--------+------------+---------------+
+    |    3 | HomRef  | ~HomRef | HomVar | Auto       | Dad, Kid      |
+    +------+---------+---------+--------+------------+---------------+
+    |    4 | ~HomRef | HomRef  | HomVar | Auto       | Mom, Kid      |
+    +------+---------+---------+--------+------------+---------------+
+    |    5 | HomRef  | HomRef  | HomVar | Auto       | Kid           |
+    +------+---------+---------+--------+------------+---------------+
+    |    6 | HomVar  | ~HomVar | HomRef | Auto       | Dad, Kid      |
+    +------+---------+---------+--------+------------+---------------+
+    |    7 | ~HomVar | HomVar  | HomRef | Auto       | Mom, Kid      |
+    +------+---------+---------+--------+------------+---------------+
+    |    8 | HomVar  | HomVar  | HomRef | Auto       | Kid           |
+    +------+---------+---------+--------+------------+---------------+
+    |    9 | Any     | HomVar  | HomRef | HemiX      | Mom, Kid      |
```

Comment:
This is my attempt to try to do the right thing for haploid calls -- We've noted that the behavior is only defined for diploid, but I think the way I've written it will work for haploid too.

Reply:
would appreciate input!

Reply:
I think adding ploidy support in January probably broke the sex chroms for mendel errors when hemiploid samples are actually ploidy 1

Reply:
this didn't appear to be documented/tested guarantees though.

Reply:
This is fine. For ploidy == 1, HomRef and HomVar are exactly what you are testing for below.